oss_src/unity/python/sframe/sys_util.py

```diff
@@ -439,3 +442,98 @@ def _get_expanded_classpath(classpath):
                     for path in classpath.split(os.path.pathsep))
     logging.getLogger(__name__).debug('classpath being used: %s' % jars)
     return jars
+
+def get_library_name():
+    """
+    Returns either sframe or graphlab depending on which library
+    this file is bundled with.
+    """
+    from os.path import split, abspath
+    
+    __lib_name = split(split(abspath(sys.modules[__name__].__file__))[0])[1]
+
+    assert __lib_name in ["sframe", "graphlab"]
+
+    return __lib_name
+
+
+def get_config_file():
+    """
+    Returns the file name of the config file from which the environment
+    variables are written.
+    """
+    import os
+    from os.path import abspath, expanduser, join, exists
+
+    __lib_name = get_library_name()
+
+    assert __lib_name in ["sframe", "graphlab"]
+
+    __default_config_path = join(expanduser("~"), ".%s" % __lib_name, "config")
+
+    if "GRAPHLAB_CONFIG_FILE" in os.environ:
+        __default_config_path = abspath(expanduser(os.environ["GRAPHLAB_CONFIG_FILE"]))
+
+        if not exists(__default_config_path):
+            print(("WARNING: Config file specified in environment variable "
+                   "'GRAPHLAB_CONFIG_FILE' as "
+                   "'%s', but this path does not exist.") % __default_config_path)
+
+    return __default_config_path
+
+
+def setup_environment_from_config_file():
+    """
+    Imports the environmental configuration settings from the
+    config file, if present, and sets the environment
+    variables to test it.
+    """
+
+    from os.path import exists
+    
+    config_file = get_config_file()
+
+    if not exists(config_file):
+        return
+    
+    config = _ConfigParser.SafeConfigParser()
```

Comment:
Do a big try except here. Just in case issues like
 - file exists but no read permissions
 - malformed file.